skt/misc.py

```diff
@@ -24,6 +25,39 @@
 LOGGER = logging.getLogger()
 
 
+@retryable(RuntimeError)
+def retry_safe_popen(err_exc_strings, *args, stdin_data=None, **kwargs):
+    """ Call safe_popen with *args, stdin_data=None, **kwargs provided,
+        If stderr stream is present and contains any string in err_exc_strings
+        list, then the process call is done again with retry after 3 seconds
+        (see retryable decorator). Log commands retry and allow 3 retries max.
+        Also log if last command failed and we gave up. The program execution
+        is not terminated / no exception is raised on last failure.
```

Comment:
This is a rather lengthy and detailed explanation. Would be great if you could begin with a single sentence giving a short and simple summary. Something like "Execute a program, retrying up to three times if certain error output is detected." Then have the detailed explanation as a separate paragraph or better two.